frontend/sw360-portlet/src/main/webapp/js/modules/codeScoop.js

```diff
@@ -0,0 +1,966 @@
+        modalBox.appendChild(modalContent);
+        _this.modalContent = modalContent;
+
+        document.body.appendChild(modalBox);
+    };
+
+    this.open = function (content, title) {
+        if (content) {
+            _this.modalTitle.innerHTML = "";
+            _this.modalContent.innerHTML = "";
+            _this.modalContent.appendChild(content);
+            if (title) {
+                _this.modalTitle.innerHTML = title;
+            }
+        } else {
+            _this.modalTitle.innerHTML = "Fetching data...";
+            _this.setLoader();
+        }
+
+        _this.modalBox.style.display = "block";
+        _this.modalOverlay = document.createElement("div");
+        _this.modalOverlay.style.zIndex = "100";
+        _this.modalOverlay.setAttribute("class", "ui-widget-overlay ui-front");
+        document.body.appendChild(_this.modalOverlay);
+        _this.opened = true;
+    };
+
+    this.setLoader = function () {
+        _this.modalTitle.innerHTML = "Fetching data...";
+
+        _this.modalContent.innerHTML = "";
+        var loader = document.createElement("img");
+        loader.style.display = "flex";
+        loader.style.width = "3em";
+        loader.style.height = "3em";
+        loader.style.margin = "4em 0";
+        loader.setAttribute("src", "/sw360-portlet/images/loader.gif");
+        _this.modalContent.appendChild(loader);
+    };
+
+    this.refresh = function (content, title) {
+        _this.modalTitle.innerHTML = "";
+        _this.modalContent.innerHTML = "";
+        _this.modalContent.appendChild(content);
+        if (title) {
+            _this.modalTitle.innerHTML = title;
+        }
+    };
+
+    this.close = function () {
+        _this.modalTitle.innerHTML = "";
+        _this.modalBox.style.display = "none";
+        _this.modalContent.innerHTML = "";
+        _this.modalOverlay.remove();
+        _this.opened = false;
+    };
+
+    this.isOpened = function () {
+        return _this.opened;
+    };
+
+    _this.init();
+}
+
+define("modules/codeScoop", [], function() {
+    function codeScoop(apiUrl, apiToken) {
+
+        var _this = this;
+
+        this.apiUrl = apiUrl;
+        this.apiToken = apiToken;
+        this.formElements = {
+            form: undefined,
+            name: undefined,
+            categories: undefined,
+            homepage: undefined,
+            description: undefined,
+            componentList: undefined,
+            componentListUrl: undefined
+        };
+        this.indexTable = undefined;
+        this.indexData = {};
+        this.timeout = undefined;
+        this.indexDone = false;
+        this.interval = 500;
+        this.modal = new Modal();
+
+        this._api = function (method, path, request, callback) {
+            var xhr = new XMLHttpRequest();
+            xhr.open(method, _this.apiUrl + path, true);
+            xhr.setRequestHeader("X-Api-Key", _this.apiToken);
+            xhr.setRequestHeader("X-User-Login", "test");
```

Comment:
is this a hard-coded user name?

Reply:
@alexbrdn its hardcoded access type for CD system. sw360 will not send any type of personal data from this script

Reply:
It's not the personal data I have concerns for. I was wondering whether this is some kind of parameter that should be parametrized rather than hard-coded. When you hard-code `"test"` it does not look like it's meant for production, you know.